neuraxle/base.py

```diff
@@ -2020,18 +2033,23 @@ def mutate(self, new_method="inverse_transform", method_to_assign_to="transform"
         else:
             return BaseStep.mutate(self, new_method, method_to_assign_to, warn)
 
-    def _step_name_to_index(self, step_name):
-        for index, (current_step_name, step) in self.steps_as_tuple:
-            if current_step_name == step_name:
-                return index
-
     def _step_index_to_name(self, step_index):
         if step_index == len(self.items()):
             return None
 
         name, _ = self.steps_as_tuple[step_index]
         return name
 
+    def __setitem__(self, key: str, new_step: BaseStep):
```

Comment:
Please support `key` of type `Union[str, int]` as like in `__setitem__`. Update docstrings and types at both places if needed.